File: src/cgi/file.py
#!/usr/bin/env python3
import os
import sys
import re
from urllib.parse import parse_qs

def parse_input():
    if "REQUEST_METHOD" in os.environ:
        if os.environ["REQUEST_METHOD"] == "POST":
            # Read the raw POST data
            content_length = int(os.environ.get("CONTENT_LENGTH", 0))
            post_data = sys.stdin.read(content_length)
            # Parse the raw POST data into key-value pairs
            pattern = r'name="([^"]+)"\s*\r?\n\r?\n([^-\s]+)'
            matches = re.findall(pattern, post_data)
            keys_and_values = {}
            for match in matches:
                key = match[0]
                value = match[1]
                keys_and_values[key] = value
            print(keys_and_values)
        elif os.environ["REQUEST_METHOD"] == "GET":
            # Parse the query string into key-value pairs
            query_string = os.environ["QUERY_STRING"]
            keys_and_values = parse_qs(query_string)
            print(keys_and_values)

if __name__ == "__main__":
    parse_input()

File: src/cgi/file.php
<?php

if ($_SERVER["REQUEST_METHOD"] === "POST") {
    $handle = fopen("php://stdin", "r");
    $body = trim(fread($handle, 1024));
    echo $body;
    fclose($handle);
    $pattern = '/name="([^"]+)"\s*\r?\n\r?\n([^-\s]+)/';
    preg_match_all($pattern, $body, $matches, PREG_SET_ORDER);

    $keysAndValues = [];


    foreach ($matches as $match) {
        $key = $match[1];
        $value = $match[2];
        $keysAndValues[$key] = $value;
    }
    print_r($keysAndValues);
}
if ($_SERVER["REQUEST_METHOD"] === "GET" ) {
    $querry = $_SERVER["QUERY_STRING"];
    // Parse the query string into an associative array
    parse_str($querry, $keysAndValues);
    // Output the extracted key-value pairs
    print_r($keysAndValues);
}
?>


